pkgs/by-name/py/pyrefly/package.nix

```diff
@@ -8,23 +8,33 @@
 }:
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "pyrefly";
-  version = "0.34.0";
+  version = "0.42.2";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "pyrefly";
     tag = finalAttrs.version;
-    hash = "sha256-HPPDsvWEFfh/GNMUPiVjQr28YBBs2DACBGM3cxo5Nx4=";
+    hash = "sha256-jNQnqiIZHSAzL/6m7AGxGuhFvlOZZLATbm5MX9aReaw=";
   };
 
+  patches = [
+    ./notebook_test.patch
+    ./shebang.patch
```

Comment:
I would suggest switching to `substituteInPlace --replace-fail` for both these one liners.